apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-common/src/main/java/co/elastic/apm/agent/esrestclient/ElasticsearchRestClientInstrumentationHelper.java

```diff
@@ -119,12 +119,17 @@ public void finishClientSpan(@Nullable Response response, Span span, @Nullable T
             int statusCode = -1;
             String address = null;
             int port = -1;
+            String cluster = null;
             if (response != null) {
                 HttpHost host = response.getHost();
                 address = host.getHostName();
                 port = host.getPort();
                 url = host.toURI();
                 statusCode = response.getStatusLine().getStatusCode();
+
+                // response header has higher priority than fetched/cached value
```

Comment:
[minor] Is it leftover, or I just didn't understand it properly?

Reply:
this is definitely a leftover, thanks for catching this !